### ISSUE TITLE
Discovery: Should Links Open in a New Tab?

### ISSUE THREAD

**Issue by jknipes** [department-of-veterans-affairs/va.gov-team#77453]
## Issue Description
Consider changing PW links to **not** open a new tab. This issue has been raised by Naomi Marcusson as a possible accessibility issue.

---
## Tasks
- [x] Team review Web Content Accessibility Standards (WCAG) in comments section
- [x] Contact CAIA: What is the current guidance regarding new tabs opening from links?
- [x] Discuss priority level with @EWashb 
- 
## Acceptance Criteria
- [x] Team has determined whether it is best to disable new tabs resulting from links on PW
- [X] Team has determined whether disabling this feature is a one-step process Scroll Viewport

## Related Ticket
[80169](https://github.com/department-of-veterans-affairs/va.gov-team/issues/80917)

## Web Content Accessibility Standards (WCAG)
[G200: Opening new windows and tabs from a link only when necessary](https://www.w3.org/TR/2016/NOTE-WCAG20-TECHS-20161007/G200)
---
## How to configure this issue
- [ ] **Attached to a Milestone** (when will this be completed?)
- [X] **Attached to an Epic** (what body of work is this a part of?)
- [ ] **Labeled with Team** (`product support`, `analytics-insights`, `operations`, `service-design`, `Console-Services`, `tools-fe`)
- [ ] **Labeled with Practice Area** (`backend`, `frontend`, `devops`, `design`, `research`, `product`, `ia`, `qa`, `analytics`, `contact center`, `research`, `accessibility`, `content`)
- [ ] **Labeled with Type** (`bug`, `request`, `discovery`, `documentation`, etc.)

**Comment by jknipes** [comment]
For internal links to open in the **same** tab, the link must go to a page in Confluence from a deployed space.

**Comment by laucon** [comment]
Two things: 
A. I looked at google analytics and it looks like there is no way to do this audit using google analytics. It's too big and complex.
B. Here is a list of steps we can do to complete this audit. Note that every 3 steps are connected. So steps 1, 2, & 3 are connected and then steps 6, 7, and 8 are connected and so on until step 19 which is separate.

1. Manually audit Getting Started docs for internal links
2. See and write down which internal links in Getting Started docs open in new tabs
3. Fix the problematic internal links in Getting Started docs
4. Manually audit Research and Design for internal links
5. See and write down which internal links in Research and Design docs open in new tabs
6. Fix the problematic internal links in Research and Design docs
7. Manually audit Analytics and Insights for internal links
8. See and write down which internal links in Analytics and Insights docs open in new tabs
9. Fix the problematic internal links in Analytics and Insights docs
10. Manually audit Collaboration Cycle docs for internal links
11. See and write down which internal links in Collaboration Cycle docs open in new tabs
12. Fix the problematic internal links in Collaboration Cycle docs
13. Manually audit Support docs for internal links
14. See and write down which internal links in Support docs open in new tabs
15. Fix the problematic internal links in Support docs
16. Manually audit Developer docs for internal links
17. See and write down which internal links in Developer docs open in new tabs
18. Fix the problematic internal links in Developer docs
19. Create new rules for how to link to internal docs and add those to a LB list or something

**Comment by laucon** [comment]
It sounds like I, Jill, and Brian have all worked on this ticket to some degree, so maybe we should assign it to the 3 of us so that we all get credit for having worked on it.

**Comment by jknipes** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Platform-Content-Team-62bde00be46b400017674aef/issues/department-of-veterans-affairs/va.gov-team/77453?planning-poker) with Zenhub @BSmolensky @taylojill @CherylEvans @laucon

**Comment by jknipes** [comment]
From ticket 80169 comments: 
For reference, here's the [WCAG guideline](https://www.w3.org/TR/2016/NOTE-WCAG20-TECHS-20161007/G200) around this.

In general, you only want to open things in a new tab when

1. You're in the middle of a form, and clicking the link would take them out of that flow; or 
2. Clicking the link would log someone out of a secure part of the website 

And if you're opening a link in a new tab, you should include "(opens in new tab)" in the link text so users are aware of that behavior.

**Comment by jknipes** [comment]
Description from Web Content Accessibility Guidelines (WCAG) :
The objective of this technique is to limit the use of links or buttons that open new windows or tabs within Web content. In general, it is better not to open new windows and tabs since they can be disorienting for people, especially people who have difficulty perceiving visual content. However there are some situations where it is preferable from an accessibility perspective to open a new window or tab. Here are two such situations:

Opening a page containing context-sensitive information, such as help instructions, or an alternate means of completing a form, such as a calendar-based date picker, will significantly disrupt a multi-step workflow, such as filling in and submitting a form, if the page is opened in the same window or tab.

The user is logged into a secured area of a site, and following a link to a page outside of the secured area would terminate the user's logon. In this case opening external links in an external window allows the user to access such references while keeping their login active in the original window.

It is recommended that when links are opened to a new window, there is advance warning.

**Comment by jknipes** [comment]
Per k15t: in order for internal links to open in the same tab, the original link must go to a page in one of the deployment spaces Confluence.
Links on Confluence pages to external websites including Platform Website will always open in a new tab when deployed.

**Comment by laucon** [comment]
I'm looking deeper at different sections. I feel like 90% of internal tabs in Research and Design open in a new tab whereas with Dveloepr Docs, it's more varied. Also, Research and Design has fewer overall docs. So it might make sense to just go into research and design and straight up fix documents one by one whereas with Developer Docs, it might make more sense to make a list of which docs have internal links and which of those docs need fixing. I hope that makes sense.

**Comment by jknipes** [comment]
CAIA Team advised in ticket [81833](https://github.com/department-of-veterans-affairs/va.gov-team/issues/81833):
> All links should default to opening in the same tab.
>
> The only exception to this, is if the link is embedded in the content a form flow or task, and opening that link would cause the user to lose their place or data they've entered. This scenario should be avoided when possible. If the link does open in a new tab, the link label should clearly indicate that.
>
> See the [links section](https://design.va.gov/content-style-guide/links) in the content style guide for information on link labels.

[Source: GitHub Decision on links by Mikki](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/1680#issuecomment-2013837291)

**Comment by laucon** [comment]
I don't know if it's possible for us to do that with all links. I think we can only do it with all internal links.

**Comment by jknipes** [comment]
Changing estimate to 8pts to capture work already done by the team on this ticket

**Comment by jknipes** [comment]
Recently announced plans by k15t to correct the behavior of **all** links has caused the team to pause work on this project: 
>Laura Parraga Gonzalez (K15t) commented:
HI Joe,
All links will open in the same tab after our improvement is implemented. No matter if they are internal, external, point to your Confluence, to the Viewport site or to any other website.
We are aiming to include the issue in our next release, hopefully at the beginning of next month.
Cheers,
Laura

**Comment by jknipes** [comment]
Discovery has been completed, Content Team will defer this work to k15t per the announcement in comments above.
